satellite-software/src/AmbaSat1App.cpp

```diff
@@ -178,7 +182,7 @@ void AmbaSat1App::loop()
     // sleep device for designated sleep cycles
     for (int i=0; i < _config.getUplinkSleepCycles(); i++)
     {
-        LowPower.powerDown(SLEEP_8S, ADC_OFF, BOD_OFF);    //sleep 8 seconds * sleepcycles
+        LowPower.powerDown(TIME_INTERVAL, ADC_OFF, BOD_OFF);    //sleep TIME_INTERVAL seconds * sleepcycles
```

Comment:
You actually need use one of [the enums defined here](https://github.com/rocketscream/Low-Power/blob/master/LowPower.h#L6) for `TIME_INTERVAL`. You, of course, could define `TIME_INTERVAL` to be one of those enums, but you cannot define it as `30` as you do above. `LowPower.powerDown()` is a wrapper around the system level call `wdt_enable` ([see here for docs](https://www.nongnu.org/avr-libc/user-manual/group__avr__watchdog.html)), which also expects a finite set of possible values from a list of enumerations. Basically, the behavior of the code you implemented here is undefined because you are passing an invalid value for the first argument. The largest sleep interval the ATMEGA328P supports is 8 seconds. You just have to sequence multiple sleep intervals if you want to sleep longer.

With that said, the way I designed this code you shouldn't be changing the sleep interval size on this line. Trace back what `_config.getUplinkSleepCycles()` does. I intend for sleep interval to be a command-able configuration, meaning you send the device a command via TTN to change the sleep interval used. Given that, your change as-is is not something I would incorporate into the master code base.

Reply:
yes, actually this: PersistedConfiguration::resetToDefaults() is the root cause of all this troubles.
I struggled a lot to find it and understand that is what i needed to modify to change the sleep cycle.
so what i tried to do is extrnalize it to a global definition that is easier to spot and use it in the constructor in a more explicit way (see     _config.setUplinkSleepCycles(DEFAULT_RATE_VALUE, true); in the setup() method).
other than that, i extrnalized the 8 seconds too, as they do work in pairs.

anyway, i realize this part of the change is messy and the default is not actually changed and i really couldn't do it, as per how the includes are organized.

so yeah, i understand if you want to reject this too

Reply:
I think you might be missing the design is doing. The function `PersistedConfiguration::resetToDefaults()` is ostensibly only called once ever across multiple reboots (reality, if you change the persisted memory bit layout, it gets called again). It saves the transmit rate to EEPROM on the ATMEGA328P. After that point, the code gets the transmit rate value from EEPROM, which you can change at run time. You should never have to change code in order to get a different transmit rate, you change the run time configuration value. to change that , you need to send a downlink to your device from the TTN web interface. The downlink is raw binary value, and I haven't yet documented it well, but you can [read more here](https://github.com/michaelkamprath/ambasat1-flight-software/tree/master/satellite-software#set-sensor-telemetry-uplink-rate). The whole idea here is so you can change the transmit rate of the satellite once it is in orbit, because once it is in orbit you cannot recompile the code. 

TL;DR: Changing the transmit rate using this code should not be done with compiler constants, but with satellite downlink commands.